### ISSUE TITLE
make the alignment of >= 128 bit integer types 16 bytes to prevent failure to properly align u128 when using 16-bit cmpchxg

### ISSUE THREAD

**Issue by andrewrk** [ziglang/zig#2987]
`@cmpxchg` on x86_64 supports u128 integers. However `@alignOf(u128)` reports 8, and without `align(16)` on the `u128`, the code segfaults at runtime. This alignment is reported by LLVM as the "ABI alignment". Is this alignment incorrect? Is the atomic alignment requirement higher? Investigation is needed.

Here's the test case:

```zig
test "128-bit cmpxchg" {
    if (builtin.arch != .x86_64) {
        return error.SkipZigTest;
    }
    var x: u128 = 1234; // segfault at runtime can be fixed by adding align(16) here
    if (@cmpxchgWeak(u128, &x, 99, 5678, .SeqCst, .SeqCst)) |x1| {
        expect(x1 == 1234);
    } else {
        @panic("cmpxchg should have failed");
    }

    while (@cmpxchgWeak(u128, &x, 1234, 5678, .SeqCst, .SeqCst)) |x1| {
        expect(x1 == 1234);
    }
    expect(x == 5678);

    expect(@cmpxchgStrong(u128, &x, 5678, 42, .SeqCst, .SeqCst) == null);
    expect(x == 42);
}
```

**Comment by gereeter** [comment]
See also rust-lang/rust#54341 describing the same issue. As mentioned there, this was [fixed in LLVM](https://reviews.llvm.org/D28990) but [reverted](https://reviews.llvm.org/rL294709). Clang [manually hardcodes](https://github.com/llvm-mirror/clang/blob/c8c0024353782e40a8711f025ed1a4091e5c0975/lib/AST/ASTContext.cpp#L1854-L1858) the alignment of 128 bit types, ignoring what LLVM reports. As brought up in that LLVM change, all investigated ABIs mandated that 128-bit types be aligned to 16 bytes.

**Comment by andrewrk** [comment]
Thanks for that @gereeter. This is now a contributor friendly issue.

Here is where the alignment of integer types is set: https://github.com/ziglang/zig/blob/6cb4cac5cd1c68f41c62a3c23b09513988337c8d/src/analyze.cpp#L5629

**Comment by andrewrk** [comment]
Will be fixed with the merge of #3033
